### PR TITLE
Add --if-missing flag to local.hex mix task

### PR DESCRIPTION
**Pull request by hairyhum**
In automation scripts like `make`, `--force` flag causes reinstallation of Hex
every time `mix` is run (to avoid hanging on prompt), which is unnecessary.
Locating that Hex is installed from make tasks is not a trivial task,
so adding `mix local.hex --if-missing` would be much more convenient.

### CODE REVIEW

**Comment by josevalim** [comment]
The problem then is that you can end-up with an outdated Hex that is never updated?

**Comment by hairyhum** [comment]
Yes, you can end up with outdated version. But this flag is supposed to be used in automation scripts that can be run multiple times, reinstalling Hex. Keeping Hex up to date is not a goal here.

**Comment by josevalim** [comment]
@hairyhum can we please update the documentation then? We should say it is intended to be used if Hex needs to be checked multiple times inside the same script. We should also say this risks Hex being out of date.

File: lib/mix/lib/mix/tasks/local.hex.ex [hunk 1]
```diff
@@ -16,13 +16,35 @@ defmodule Mix.Tasks.Local.Hex do
...
   If you want to change the [default mirror](https://repo.hex.pm)
   used for fetching Hex, set the `HEX_MIRROR` environment variable.
   """
   @spec run(OptionParser.argv) :: boolean
-  def run(args) do
+  def run(argv) do
+    {opts, _} = OptionParser.parse!(argv, switches: [if_missing: :boolean,
+                                                     force: :boolean])
+    should_install = case opts do
+      %{force: true}      -> true;
```
Comment (josevalim) [inline lib/mix/lib/mix/tasks/local.hex.ex]: Are you sure this works? AFAIK `opts` is a keyword list and not a map.

**Comment by hairyhum** [comment]
By "out of date" do you mean failing `Mix.Hex.ensure_updated?` check?
If so, it can be checked the same way to have the required version.

File: lib/mix/lib/mix/tasks/local.hex.ex [hunk 2]
```diff
@@ -16,13 +16,37 @@ defmodule Mix.Tasks.Local.Hex do
...
   @spec run(OptionParser.argv) :: boolean
-  def run(args) do
+  def run(argv) do
+    {opts, _} = OptionParser.parse!(argv, switches: [if_missing: :boolean,
+                                                     force: :boolean])
+    force = opts[:force] || false
+    if_missing = opts[:if_missing] || false
+
+    should_install = case {force, if_missing} do
+      {false, true} -> Code.ensure_loaded?(Hex);
```
Comment (ericmj) [inline lib/mix/lib/mix/tasks/local.hex.ex]: Trailing `;` is not needed.

File: lib/mix/lib/mix/tasks/local.hex.ex [hunk 3]
```diff
@@ -16,13 +16,37 @@ defmodule Mix.Tasks.Local.Hex do
...
+      _             -> true
```
Comment (ericmj) [inline lib/mix/lib/mix/tasks/local.hex.ex]: We usually don't align these to make future diffs better.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**